packages/forms/backend/sirius-components-widget-reference/src/main/java/org/eclipse/sirius/components/widget/reference/ReferenceWidgetComponent.java

```diff
@@ -114,7 +114,7 @@ public Element render() {
                 .label(label)
                 .iconURL(iconURL)
                 .values(items)
-                .options(options)
+                .optionsProvider(() -> options)
```

Comment:
`options` is still computed on render, we are just deferring the transfer of the values to the front on demand. Are we sure this is enough to fix the performance issues?

Reply:
If we just do it this way, I think we could simply keep the ReferenceWidgetComponent & widget as before, just not declare the `referenceOptions: [ReferenceValue!]!` in the GraphQL schema so that the front is required to use the separate query if/when it needs them.

Reply:
Yes, you're right, I tested it with a big model and there is some latency on each render.
I'll change it to compute the options inside the supplier.

Reply:
Done